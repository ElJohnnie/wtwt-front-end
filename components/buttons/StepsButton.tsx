import React from 'react';

export interface StepsButtonProps {
  icon: JSX.Element;
  callback?: () => void;
}

export default function StepsButton(props: StepsButtonProps) {
  const { icon, callback } = props;
  return (
    <div className="flex flex-row justify-center items-center">
      <button
        className="w-24 h-24 rounded-full  
          inline-flex items-center justify-center  
          text-white text-[24px] font-bold bg-pink-600 hover:bg-pink-400"
        onClick={callback}
      >
        {icon}
      </button>
    </div>
  );
}
