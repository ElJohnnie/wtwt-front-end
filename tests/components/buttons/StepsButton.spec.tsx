import React from 'react';
import { render } from '@testing-library/react';
import StepsButton from '../../../components/buttons/StepsButton';

describe('StepsButton', () => {
  it('quero renderizar o StepsButton inicialmente', () => {
    const { container } = render(
      <StepsButton
        icon={
          <svg
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            strokeWidth={1.5}
            stroke="currentColor"
            className="size-6"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
            />
          </svg>
        }
      />,
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
