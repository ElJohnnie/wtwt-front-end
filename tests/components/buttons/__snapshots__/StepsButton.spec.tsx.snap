// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepsButton quero renderizar o StepsButton inicialmente 1`] = `
<div>
  <div
    class="w-full flex flex-col justify-center items-center"
  />
</div>
`;
